### PR TITLE
 Suppress favicon 404 in default html pages

### PR DESCRIPTION
**Pull request by VirtualTim**
Minor change, but this should stop the favicon 404 and console errors.

### CODE REVIEW

**Comment by kripken** [comment]
Looks good, although this does increase code size - @juj, didn't you recently add a test to check for changes, which i think this change should cause?

@VirtualTim, please add yourself to AUTHORS, btw.

**Comment by kripken** [comment]
Oh, and yeah, I did see those console errors myself - nice to fix those!

File: AUTHORS [hunk 1]
```diff
@@ -385,5 +385,7 @@ a license to everyone to use it as detailed in LICENSE.)
 * Florian Stellbrink <florian@stellbr.ink>
 * Shane Peelar <lookatyouhacker@gmail.com>
 * Alessandro Pignotti <alessandro@leaningtech.com>
+* Martina Kraus <kraus.martina.m@googlemail.com>
+* Tim Lander <tim57282＠hotmail.com>
 * Zheng Tao Lee <zhengtao.lee@autodesk.com> (copyright owned by Autodesk, Inc.)
-* Martina Kraus <kraus.martina.m@googlemail.com>
\ No newline at end of file
+* Martina Kraus <kraus.martina.m@googlemail.com>
```
Comment (juj) [inline AUTHORS]: Hmm, this now has a duplicate line?

**Comment by juj** [comment]
> Looks good, although this does increase code size - @juj, didn't you recently add a test to check for changes, which i think this change should cause?

This change does not apply to `shell_minimal_runtime.html`, so it does not get included to `other.test_minimal_runtime_code_size`. Also, I do not think that this change should be present in that file either, it should contain the least amount of necessary things to educate how a developer will set up a loader shell. Adding favicon handling there may confuse some to think that it might somehow be necessary for proper behavior.

**Comment by kripken** [comment]
Hmm, good point @juj - maybe this might be confusing to include, actually, the same logic as for the minimal html basically (although there it's even more crucial since every line should be meaningful). So not sure how we should go here.

**Comment by VirtualTim** [comment]
>  This change does not apply to `shell_minimal_runtime.html` [...] Also, I do not think that this change should be present in that file either, it should contain the least amount of necessary things to educate how a developer will set up a loader shell.

Yep, that was my reasoning too.
I did include it in `shell_minimal.html`, as that already includes things like a title.

I think if we're producing sample html files it makes sense to have these files produce no errors or warnings (during standard operation), right? Even if they're just a favicon 404.

File: AUTHORS (hunk 1 above)
Comment (VirtualTim) [reply AUTHORS]: Good spot. I'll remove the dupe.

**Comment by kripken** [comment]
Good point about avoiding warnings - another option is to emit a favicon file, I guess.

Another worry came to me - this would prevent a user's favicon from being shown, unless they customize the HTML?

**Comment by VirtualTim** [comment]
For my own personal fork of the shell I'm using a base64 encoded icon embedded in the html. I could whip something like that up, if you think that's better? However I think that it makes the most sense to add the smallest amount of html to avoid the spurious request.

I guess it _could_ prevent a users favicon from showing up, if they were just dumping an icon in the root folder. However I think that anyone who cares enough to add an icon would add it to the html. And my change is basically acting as a placeholder for that.

**Comment by kripken** [comment]
I see what you mean, but I do worry this may surprise some people. As the benefit is just removing a warning, I'm not sure it's worth doing for that reason. So I lean towards not merging this, sorry.

(Regardless of if we end up not merging this, thanks for submitting it.)

**Comment by VirtualTim** [comment]
No worries. It's such a minor change it's not worth putting too much though into it.
I won't close it, in case you change your mind, but feel free to reject it.

**Comment by kripken** [comment]
Ok, let's close this - I wanted to wait a bit and see if other people had another perspective here. Seems like no, so as we agreed it seems safest to not merge this. Thanks again though for submitting it!

**Pull request closed without merging**